graphql_compiler/schema_generation/schema_graph.py

```diff
@@ -165,34 +136,105 @@ def class_fields(self):
     @property
     def is_vertex(self):
         """Return True if the schema element represents a vertex type, and False otherwise."""
-        return self._kind == SchemaElement.ELEMENT_KIND_VERTEX
+        return isinstance(self, Vertex)
 
     @property
     def is_edge(self):
         """Return True if the schema element represents an edge type, and False otherwise."""
-        return self._kind == SchemaElement.ELEMENT_KIND_EDGE
+        return isinstance(self, Edge)
 
     @property
     def is_non_graph(self):
         """Return True if the schema element represents a non-graph type, and False otherwise."""
-        return self._kind == SchemaElement.ELEMENT_KIND_NON_GRAPH
+        return isinstance(self, NonGraphElement)
+
+    @abstractmethod
+    def __str__(self):
+        pass
+
+    @abstractmethod
+    def __repr__(self):
+        pass
+
+
+class GraphElement(SchemaElement):
+    def __init__(self, class_name, abstract, properties, class_fields):
+        super(GraphElement, self).__init__(class_name, abstract, properties, class_fields)
+
+        # In the schema graph, both vertices and edges are represented with vertices.
+        # These dicts have the name of the adjacent schema vertex in the appropriate direction.
+        #
+        # For vertex classes:
+        #   in  = the edge is attached with its head / arrow side
+        #   out = the edge is attached with its tail side
+        #
+        # For edge classes:
+        #   in  = the tail side of the edge
+        #   out = the head / arrow side of the edge
+        #
+        # For non-graph classes, these properties are always empty sets.
+        self.in_connections = set()
+        self.out_connections = set()
 
     def freeze(self):
         """Make the SchemaElement's connections immutable."""
         self.in_connections = frozenset(self.in_connections)
         self.out_connections = frozenset(self.out_connections)
 
+
+class Vertex(GraphElement):
```

Comment:
I'm a bit worried that `Vertex` and `Edge` might get confused with the actual data vertices and edges. Maybe `VertexType/EdgeType` or something like that?